python/ray/serve/utils.py

```diff
@@ -180,13 +180,22 @@ def format_actor_name(actor_name, controller_name=None, *modifiers):
 
 def get_conda_env_dir(env_name):
     """Given a environment name like `tf1`, find and validate the
-    corresponding conda directory.
+    corresponding conda directory. Untested on Windows.
     """
     conda_prefix = os.environ.get("CONDA_PREFIX")
     if conda_prefix is None:
-        raise ValueError(
-            "Serve cannot find environment variables installed by conda. " +
-            "Are you sure you are in a conda env?")
+        # The caller is neither in a conda env or in (base).  This is rare
+        # because by default, new terminals start in (base), but we can still
+        # support this case.
+        conda_exe = os.environ.get("CONDA_EXE")
+        if conda_exe is None:
+            raise RayServeException(
+                "Ray Serve cannot find environment variables set by conda. "
+                "Please verify conda is installed.")
+        # Example: CONDA_EXE=$HOME/anaconda3/bin/python
+        # Strip out the /bin/python by going up two parent directories.
+        conda_prefix = os.path.abspath(
+            os.path.join(conda_exe, f"{os.pardir}/{os.pardir}"))
```

Comment:
With all these complexity of managing path. it might be worth using https://docs.python.org/3/library/pathlib.html to make this function more readable. 
```
from pathlib import Path
str(Path(conda_exe).parent.parent)
```